conans/client/cmd/uploader.py

```diff
@@ -306,6 +310,9 @@ def _compress_recipe_files(self, ref):
         src_files, src_symlinks = gather_files(export_src_folder)
         the_files = _compress_recipe_files(files, symlinks, src_files, src_symlinks, export_folder,
                                            self._output)
+        with self._cache.package_layout(ref).update_metadata() as metadata:
```

Comment:
Maybe move these lines to the `_upload_recipe`? That way would be more similar to the `_upload_package`